src/test/java/org/springframework/data/elasticsearch/core/CriteriaQueryMappingUnitTests.java

```diff
@@ -109,6 +108,158 @@ void shouldMapNamesAndConvertValuesInCriteriaQuery() throws JSONException {
+		String expected = "{\n" +
+				"  \"bool\" : {\n" +
+				"    \"should\" : [\n" +
+				"      {\n" +
+				"        \"bool\" : {\n" +
+				"          \"must\" : [\n" +
+				"            {\n" +
+				"              \"range\" : {\n" +
+				"                \"birth-date\" : {\n" +
+				"                  \"from\" : \"09.11.1989\",\n" +
+				"                  \"to\" : \"09.11.1990\",\n" +
+				"                  \"include_lower\" : true,\n" +
+				"                  \"include_upper\" : true,\n" +
+				"                  \"boost\" : 1.0\n" +
+				"                }\n" +
+				"              }\n" +
+				"            }\n" +
+				"          ],\n" +
+				"          \"adjust_pure_negative\" : true,\n" +
+				"          \"boost\" : 1.0\n" +
+				"        }\n" +
+				"      },\n" +
+				"      {\n" +
+				"        \"bool\" : {\n" +
+				"          \"must\" : [\n" +
+				"            {\n" +
+				"              \"query_string\" : {\n" +
+				"                \"query\" : \"28.12.2019\",\n" +
+				"                \"fields\" : [\n" +
+				"                  \"birth-date^1.0\"\n" +
+				"                ],\n" +
+				"                \"type\" : \"best_fields\",\n" +
+				"                \"default_operator\" : \"and\",\n" +
+				"                \"max_determinized_states\" : 10000,\n" +
+				"                \"enable_position_increments\" : true,\n" +
+				"                \"fuzziness\" : \"AUTO\",\n" +
+				"                \"fuzzy_prefix_length\" : 0,\n" +
+				"                \"fuzzy_max_expansions\" : 50,\n" +
+				"                \"phrase_slop\" : 0,\n" +
+				"                \"escape\" : false,\n" +
+				"                \"auto_generate_synonyms_phrase_query\" : true,\n" +
+				"                \"fuzzy_transpositions\" : true,\n" +
+				"                \"boost\" : 1.0\n" +
+				"              }\n" +
+				"            }\n" +
+				"          ],\n" +
+				"          \"adjust_pure_negative\" : true,\n" +
+				"          \"boost\" : 1.0\n" +
+				"        }\n" +
+				"      }\n" +
+				"    ],\n" +
+				"    \"adjust_pure_negative\" : true,\n" +
+				"    \"boost\" : 1.0\n" +
+				"  }\n" +
+				"}"; //
+
+		mappingElasticsearchConverter.updateQuery(criteriaQuery, Person.class);
+		String queryString = new CriteriaQueryProcessor().createQuery(criteriaQuery.getCriteria()).toString();
+
+		assertEquals(expected, queryString, false);
+	}
+
+	@Test
+		// DATAES-1668
+	void shouldMapNamesAndConvertValuesInCriteriaQueryForSubCriteriaWithDate() throws JSONException {
+		Calendar calendar = Calendar.getInstance();
+		//calendar.set(1982, Calendar.FEBRUARY, 28, 13, 2, 1 );
+		calendar.setTimeInMillis(383745721653L);
+		// use POJO properties and types in the query building
+		CriteriaQuery criteriaQuery = new CriteriaQuery( //
+				Criteria.or()
+						.subCriteria( Criteria.where("birthDate") //
+								.between(LocalDate.of(1989, 11, 9), LocalDate.of(1990, 11, 9)) ) //
+						.subCriteria( Criteria.where( "createdDate").is(calendar.getTime()) ) //
```

Comment:
```suggestion
						.subCriteria( Criteria.where( "createdDate").is(383745721653L) ) //
```
no need for the `Calendar` stuff here (I'd not use the old `java.util` date classes anyway, but always the classes from `java.time`).